### PR TITLE
Add MySQL 8.1 to supported versions.

### PR DESCRIPTION
**Pull request by potiuk**
Anticipating Lazy Consensus to be reached we add 8.1 version of MySQL to supported versions.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -92,7 +92,7 @@ Apache Airflow is tested with:
 | Platform    | AMD64/ARM64(\*)        | AMD64/ARM64(\*)           |
 | Kubernetes  | 1.24, 1.25, 1.26, 1.27 | 1.24, 1.25, 1.26, 1.27    |
 | PostgreSQL  | 11, 12, 13, 14, 15     | 11, 12, 13, 14, 15        |
-| MySQL       | 5.7, 8                 | 5.7, 8                    |
+| MySQL       | 5.7, 8, 8.1            | 5.7, 8                    |
```
Comment (potiuk) [inline README.md]: To be updated when we cherry-pick that one to 2.7.1

Comment (Taragolis) [reply README.md]: What do you think if we also replace `8` -> `8.0` ?

Comment (potiuk) [inline README.md]:
```suggestion
| MySQL       | 5.7, 8.0, 8.1          | 5.7, 8.0                  |
```

File: docs/apache-airflow/installation/prerequisites.rst [hunk 2]
```diff
@@ -25,7 +25,7 @@ Airflow™ is tested with:
 * Databases:
 
   * PostgreSQL: 11, 12, 13, 14, 15
-  * MySQL: 5.7, 8
+  * MySQL: 5.7, 8, 8.1
```
Comment (potiuk) [inline docs/apache-airflow/installation/prerequisites.rst]:
```suggestion
  * MySQL: 5.7, 8.0, 8.1
```

File: README.md (hunk 1 above)
Comment (potiuk) [reply README.md]: Good point. Updated

**Review by Taragolis — APPROVED** [review]

**Pull request merged**